### PR TITLE
docs: Restructure and deduplicate security docs

### PR DESCRIPTION
**Pull request by bmorelli25**
## What does this PR do?

This PR restructures the security documentation in a few different ways:
* **Secure** content is now sorted in the order a user would complete each task.
* The content in _Configure authentication credentials_, _Configure {beatname} to use encrypted connections_, and _Secure communication with Elasticsearch_ has been consolidated and deduplicated into a single file: _Secure communication with Elasticsearch_. It has also been restructured to be more of a user guide.
* User creation has moved to individual pages to allow room for APM Server to document the creation of additional user roles.

**Before and after:**
Screen Shot 2020-05-15 at 12 27 46 PM

## What doesn't this PR do?

This PR does not touch the _Secure communication with Logstash_ or the _Use Linux Secure Computing Mode (seccomp)_ files.

## Related issues

https://github.com/elastic/apm-server/issues/3596.

## HTML preview

http://beats_18594.docs-preview.app.elstc.co/diff

## Redirects

The following pages have been removed and will need redirects:
* `/beats-tls.html`
* `/securing-beats.html `
* `/beats-basic-auth.html`

### CODE REVIEW

**Comment by bmorelli25** [comment]
I have no idea which team label to apply to this. I guess APM, as that was the initial motivation for these changes.

File: libbeat/docs/https.asciidoc [hunk 1]
```diff
@@ -10,66 +10,150 @@
...
+<2> This example shows a hard-coded password, but you should store sensitive
+values
+ifndef::serverless[]
+in the <<keystore,secrets keystore>>.
+endif::[]
+ifdef::serverless[]
+in environment variables.
+endif::[]
+endif::apm-server[]
+--
```
Comment (bmorelli25) [inline libbeat/docs/https.asciidoc:148]: Should this content move to its own top-level heading/page? The TOC would then include:

* Secure communication with Elasticsearch
* Secure communication with Kibana
* Secure communication with Logstash

Alternatively, we could group these under a new header, like this:
* Secure communication with the Elastic Stack
  * Elasticsearch
  * Kibana
  * Logstash

File: libbeat/docs/security/users.asciidoc [hunk 2]
```diff
@@ -23,9 +23,16 @@ subset of the privileges needed by {beatname_uc} users. When possible, use the
 built-in roles to minimize the affect of future changes on your security
 strategy.
 
+Instead of using usernames and passwords, roles and privileges can be assigned to
+API keys to grant access to Elasticsearch resources. See <<beats-api-keys>> for
+more information
 
 [[privileges-to-setup-beats]]
-==== Grant privileges and roles needed for setup
+=== Grant privileges and roles needed for setup
```
Comment (bmorelli25) [inline libbeat/docs/security/users.asciidoc:31]: If it's alright, I'll probably move each role into its own file in `libbeat/docs/security/`.

**Comment by bmorelli25** [comment]
Please hold on reviewing this. I have some links to fix.

**Comment by bmorelli25** [comment]
Ready for review again.
```
14:11:24 INFO:build_docs:Bad cross-document links:
14:11:24 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elasticsearch/reference/master/beats.html:
14:11:24 INFO:build_docs:   - en/beats/auditbeat/master/securing-beats.html
14:11:24 INFO:build_docs:   - en/beats/filebeat/master/securing-beats.html
14:11:24 INFO:build_docs:   - en/beats/heartbeat/master/securing-beats.html
14:11:24 INFO:build_docs:   - en/beats/metricbeat/master/securing-beats.html
14:11:24 INFO:build_docs:   - en/beats/packetbeat/master/securing-beats.html
14:11:24 INFO:build_docs:   - en/beats/winlogbeat/master/securing-beats.html
14:11:24 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elasticsearch/reference/master/ccs-clients-integrations.html:
14:11:24 INFO:build_docs:   - en/beats/auditbeat/master/securing-beats.html
14:11:24 INFO:build_docs:   - en/beats/filebeat/master/securing-beats.html
14:11:24 INFO:build_docs:   - en/beats/heartbeat/master/securing-beats.html
14:11:24 INFO:build_docs:   - en/beats/metricbeat/master/securing-beats.html
14:11:24 INFO:build_docs:   - en/beats/packetbeat/master/securing-beats.html
14:11:24 INFO:build_docs:   - en/beats/winlogbeat/master/securing-beats.html
14:11:24 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elasticsearch/reference/master/get-started-verify-users.html:
14:11:24 INFO:build_docs:   - en/beats/filebeat/master/securing-beats.html
14:11:24 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elasticsearch/reference/master/ssl-tls.html:
14:11:24 INFO:build_docs:   - en/beats/filebeat/master/securing-beats.html
```

Broken links have been fixed in https://github.com/elastic/elasticsearch/pull/56875.

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: libbeat/docs/https.asciidoc (hunk 1 above)
Comment (EamonnTP) [reply libbeat/docs/https.asciidoc]: I do like the idea of grouping the content under a new heading, "Secure communication with the Elastic Stack".

File: libbeat/docs/https.asciidoc [hunk 3]
```diff
@@ -10,66 +10,152 @@
...
++
+--
 ["source","yaml",subs="attributes,callouts"]
 ----------------------------------------------------------------------
 output.elasticsearch:
-  username: {beatname_lc} <1>
-  password: verysecret <2>
-  protocol: https <3>
-  hosts: ["elasticsearch.example.com:9200"] <4>
+  hosts: ["localhost:9200"]
```
Comment (dedemorton) [inline libbeat/docs/https.asciidoc:29]: If you remove `protocol`, might be good to show https:// in the hosts setting. I'd also use something other than localhost (though I see we do that elsewhere).

File: libbeat/docs/https.asciidoc [hunk 4]
```diff
@@ -10,66 +10,152 @@
...
++
+--
 ["source","yaml",subs="attributes,callouts"]
 ----------------------------------------------------------------------
 output.elasticsearch:
-  username: {beatname_lc}
-  password: verysecret
-  protocol: https
-  hosts: ["elasticsearch.example.com:9200"]
+  hosts: ["localhost:9200"]
```
Comment (dedemorton) [inline libbeat/docs/https.asciidoc:95]: same comment as before about including https. I guess it might be the default now?

**Pull request merged**